test_pytest_html.py

```diff
@@ -25,6 +25,23 @@ def run(testdir, path='report.html', *args):
     return result, html
 
 
+def assert_summary_filter(html, test_outcome='Passed', test_outcome_number=1):
+    cb_active = ("<input checked=\"true\""
+                 " data-test-result=\"{0}\""
+                 .format(test_outcome))
+    cb_unactive = ("<input disabled=\"true\"")
+    table_test_result_outcome = ("tbody class=\"{0} "
+                                 .format(test_outcome))
+    if(test_outcome_number != 0):
+        assert str(re.search(cb_active, html).group()) == cb_active
+        assert len(
+                    re.findall(
+                               table_test_result_outcome,
+                               html)) == test_outcome_number
```

Comment:
This is a worthwhile check, but it's not really related to the filter. Perhaps move it into `assert_summary`.